extensions/interactions/MathEquationInput/directives/math-equation-input-rules.service.ts

```diff
@@ -70,6 +70,45 @@ export class MathEquationInputRulesService {
     }
   }
 
+  MatchesUpToTrivialManipulations(
+      answer: MathEquationAnswer,
+      inputs: MathEquationRuleInputsWithSide): boolean {
+    let algebraicRulesService = new AlgebraicExpressionInputRulesService();
+
+    let positionOfTerms = inputs.y;
+
+    let splitAnswer = answer.split('=');
+    let lhsAnswer = splitAnswer[0], rhsAnswer = splitAnswer[1];
+
+    let splitInput = inputs.x.split('=');
+    let lhsInput = splitInput[0], rhsInput = splitInput[1];
+
+    if (positionOfTerms === 'lhs') {
+      return algebraicRulesService.MatchesUpToTrivialManipulations(
+        lhsAnswer, {x: lhsInput});
+    } else if (positionOfTerms === 'rhs') {
+      return algebraicRulesService.MatchesUpToTrivialManipulations(
+        rhsAnswer, {x: rhsInput});
+    } else if (positionOfTerms === 'both') {
+      return (
+        algebraicRulesService.MatchesUpToTrivialManipulations(
+          lhsAnswer, {x: lhsInput}) && (
+          algebraicRulesService.MatchesUpToTrivialManipulations(
+            rhsAnswer, {x: rhsInput})));
+    } else {
+      // Position of terms is irrelevant. So, we bring all terms on one side
+      // and perform an exact match.
+      let rhsAnswerModified = nerdamer(rhsAnswer).multiply('-1').text();
+      let expressionAnswer = nerdamer(rhsAnswerModified).add(lhsAnswer).text();
```

Comment:
Maybe we should do this addition 'structurally', without simplifying? (See comment in test case above.)

Reply:
Addressed [above](https://github.com/oppia/oppia/pull/14977#discussion_r822921379).